analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd

```diff
@@ -0,0 +1,403 @@
+
+```{r}
+# look at immune cell types vs. unknown vs. other 
+all_results_df <- all_results_df |> 
+  dplyr::mutate(
+    # first get a column that is just immune, unknown, or other
+    immune_category = dplyr::case_when(
+      consensus_annotation %in% immune_types ~ "immune",
+      consensus_annotation == "Unknown" ~ "Unknown",
+      .default = "non-immune"
+    ),
+    library_id = as.character(library_id) #remove existing levels 
+  )
+```
+
+```{r}
+# now sort just by immune percentage
+immune_pct_df <- all_results_df |> 
+  dplyr::filter(immune_category == "immune") |> 
+  dplyr::mutate(annotation_percent_immune = total_cells_per_annotation/total_cells_per_library) |>
+  dplyr::group_by(library_id) |> 
+  dplyr::summarize(
+    percent_immune = sum(annotation_percent_immune)
+  ) 
+
+library_levels <- immune_pct_df |> 
+  dplyr::arrange(desc(percent_immune)) |> 
+  dplyr::pull(library_id)
+
+# append any libraries that have 0 immune cells annotated
+non_annotated_libs <- setdiff(unique(all_results_df$library_id), library_levels)
+library_levels <- c(library_levels, non_annotated_libs)
+
+immune_results_df <- all_results_df |> 
+  dplyr::left_join(immune_pct_df, by = c("library_id")) |> 
+  dplyr::mutate(
+    percent_immune = dplyr::if_else(is.na(percent_immune), 0, percent_immune),
+    library_id = forcats::fct_relevel(library_id, library_levels)
+  )
+
+stacked_barchart(immune_results_df, fill_column = "immune_category", celltype_colors = three_color_scheme)
+```
+Let's see how the percentage of immune cells compares across projects and diagnosis. 
+
+```{r}
+stacked_barchart(immune_results_df, fill_column = "immune_category", facet_variable = "project_id", celltype_colors = three_color_scheme)
+```
+
+```{r}
+stacked_barchart(immune_results_df, fill_column = "immune_category", facet_variable = "grouped_diagnosis", celltype_colors = three_color_scheme)
+```
+
+This isn't super interesting to me since most classified cells are immune cells and there doesn't seem to be any major differences across disease types outside of ATRT having a lower percentage of classified cells than other projects. 
+
+## Macrophage vs. T cell populations 
+
+It looks like there may be some differences between T cell and macrophage infiltrate based on the tumor type. 
+Here we will look specifically at all T cells and anything in the macrophage/monocyte lineage. 
+I'm including any cell types that are in our list of possible cell types and a child of [mononuclear phagocyte](https://www.ebi.ac.uk/ols4/ontologies/cl/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FCL_0000113?lang=en) in the macro/mono group. 
+We should also note that "macrophage" here probably refers to microglia instead. 
+
+```{r}
+# t cell subtypes
+t_cells <- unique_celltypes |> 
+  stringr::str_subset("T")
+
+# macro/mono lineage 
+macro_mono_cells <- c(
+  "macrophage", 
+  "monocyte",
+  "mononuclear phagocyte", 
+  "dendritic cell"
```

Comment:
Could you only get a myeloid dendritic cell out of the consensus labels? <https://pmc.ncbi.nlm.nih.gov/articles/PMC5032838/>